.github/workflows/documentation.yaml

```diff
@@ -1,27 +1,61 @@
-name: Generate terraform docs
+name: generate-terraform-docs
 
 on:
-  pull_request:
+  push:
     branches:
       - main
     paths:
       - '**.tf'
-      - '**.md'
+
 jobs:
-  docs:
+  tf-docs:
+    name: TF docs
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-      with:
-        ref: ${{ github.event.pull_request.head.ref }}
+      - uses: actions/checkout@v3
+        id: actions-checkout
+        with:
+          ref: main
+
+      - id: git-branch
```

Comment:
Creating and pushing the branch could trigger the execution of other actions (including this one? recursively?)

Perhaps we should only create the branch if there are changes to the terraform-docs after the next action runs.

Alternatively, this workflow runs within a PR branch, adding commits as needed: https://github.com/equinix-labs/metal-java/blob/main/.github/workflows/auto-commit-client-updates.yml

Reply:
Apparently the git resources created with the default GITHUB_TOKEN cannot trigger other workflows, to prevent us from accidentally creating recursive workflow. https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs

It's deleting remote and local branch in last step if there's no changes

```
- name: Delete branch
        if: steps.terraform-docs.outputs.num_changed == '0'
 ```
       
However, I just realized that the "Create Pull Request" step can manage the creation of a new branch but itself so I will update the job to make it simple.

About that alternative, we cannot commit to a PR created from a fork.

Reply:
Updated in https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/pull/27/commits/404d577bd542e6376f1f16b4626f0a7cc7d31a07